apps/frontend/src/contexts/query-client.tsx

```diff
@@ -0,0 +1,49 @@
+import { QueryClient, QueryClientProvider as TanStackQueryClientProvider } from "@tanstack/react-query"
+import { ReactNode, useState } from "react"
+
+function makeQueryClient() {
+  return new QueryClient({
+    defaultOptions: {
+      queries: {
+        // Data is considered fresh for 30 seconds
+        staleTime: 30 * 1000,
+        // Refetch on window focus for fresh data
+        refetchOnWindowFocus: true,
+        // Don't retry on error by default (we handle errors explicitly)
+        retry: false,
+      },
+    },
+  })
+}
+
+let browserQueryClient: QueryClient | undefined = undefined
+
+function getQueryClient() {
+  // Server: always make a new query client
+  if (typeof window === "undefined") {
+    return makeQueryClient()
+  }
```

Comment:
I don't think we need server side handling? The app should never be server side rendered. We'll do an Electron app for that instead.